### PR TITLE
Feature/gal 734

### PR DESCRIPTION
**Pull request by andrew-cleveland**
## Description
Changed the Lambda config to use environmental variables rather than hardcoded url's

## Context
Without this change the url's would need to be changed manually each time the api gateway was re-deployed.

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would change existing functionality)
- [] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by Abdul-NHS — APPROVED** [review]
lgtm, good use of external css style

**Review by umayrsNHS — DISMISSED** [review]

**Review by umayrsNHS — APPROVED** [review]

File: scripts/pipeline/nonprod_phlebotomy_site_load/nonprod_phlebotomy_site_load.py [hunk 1]
```diff
@@ -118,12 +119,60 @@ def create_data_set(table_name):
...
-                    },
-                    "TableName": table_name,
+                        'Address': {
+                            'S': f'{str(address)}'
+                        },
+                        'Directions': {
+                            'S': f'{str(directions)}'
+                        },
+                        'ICBCode': {
+                            'S': f'{str(icd_code)}'
```
Comment (Abdul-NHS) [inline scripts/pipeline/nonprod_phlebotomy_site_load/nonprod_phlebotomy_site_load.py:129]: Are we duplicating what is already defined above re. address, directions etc?

File: scripts/pipeline/nonprod_population_load/nonprod_population_load.py [hunk 2]
```diff
@@ -85,8 +86,70 @@ def format_dynamodb_json(csvreader, table_name):
...
                                 },
-                                "Invited": {"S": f"{Invited}"},
-                                "LsoaCode": {"S": f"{lsoa_2011}"},
+                                'family_name': {
+                                    'S': f'{family_name}'
+                                },
+                                'date_of_birth': {
+                                    'S': f'{date_of_birth}'
+                                },
+                                'gender_code': {
```
Comment (Abdul-NHS) [inline scripts/pipeline/nonprod_population_load/nonprod_population_load.py:95]: Same goes for this section, seems like there is duplication for the other part of the code that is linted and appended

File: scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ export const writeFile = (filename, obj) => {
 };
 
 const lsoaHeader =
-  "POSTCODE,POSTCODE_2,LOCAL_AUT_ORG,NHS_ENG_REGION,SUB_ICB,CANCER_REGISTRY,EASTING_1M,NORTHING_1M,LSOA_2011,MSOA_2011,CANCER_ALLIANCE,ICB,OA_2021,LSOA_2021,MSOA_2021,IMD_RANK,IMD_DECILE,AVG_EASTING,AVG_NORTHING";
+  "POSTCODE,POSTCODE_2,LOCAL_AUT_ORG,NHS_ENG_REGION,SUB_ICB,CANCER_REGISTRY,EASTING_1M,NORTHING_1M,LSOA_2011,MSOA_2011,CANCER_ALLIANCE,ICB,OA_2021,LSOA_2021,MSOA_2021,IMD_RANK,IMD_DECILE,LSOA_NAME,AVG_EASTING,AVG_NORTHING";
```
Comment (Abdul-NHS) [inline scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.js:50]: I think this change is already in main so it shouldn't be included in the pr

File: scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.test.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ describe("Test Happy path of midpoint generation", () => {
       MSOA_2021: "E02004940",
       IMD_RANK: "30070",
       IMD_DECILE: "10",
+      LSOA_NAME: "St Albans 017E",
```
Comment (Abdul-NHS) [inline scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.test.js:23]: Same goes for here, already in main so shouldn't be included here

File: scripts/test_data/updateForecastUptakeMod/updateForecastUptakeMod.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import { Readable } from "stream";
```
Comment (Abdul-NHS) [inline scripts/test_data/updateForecastUptakeMod/updateForecastUptakeMod.js:1]: Merged into main already so shouldnt be included as changes to be added to main

File: scripts/test_data/updateForecastUptakeMod/updateForecastUptakeMod.test.js [hunk 6]
```diff
@@ -0,0 +1,208 @@
+import { fixDecimal, match } from "./utils/helper";
```
Comment (Abdul-NHS) [inline scripts/test_data/updateForecastUptakeMod/updateForecastUptakeMod.test.js:1]: Same here for this file

File: terraform/main.tf [hunk 7]
```diff
@@ -452,6 +465,81 @@ module "participants_in_lsoa_cloudwatch" {
   retention_days       = 14
 }
 
+# Calculate number of participants to invite
+module "calculate_number_to_invite_lambda" {
+  source               = "./modules/lambda"
+  bucket_id            = module.s3_bucket.bucket_id
+  lambda_iam_role      = module.iam_galleri_lambda_role.galleri_lambda_role_arn
+  lambda_function_name = "calculateNumberToInviteLambda"
```
Comment (Abdul-NHS) [inline terraform/main.tf:473]: This should already be in main, as it the lambda linked to invitations

File: terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js [hunk 8]
```diff
@@ -0,0 +1,185 @@
+import { DynamoDBClient, GetItemCommand } from "@aws-sdk/client-dynamodb";
```
Comment (Abdul-NHS) [inline terraform/src/calculateNumberToInviteLambda/calculateNumberToInviteLambda.js:1]: This shouldn't be a part of the pr as it is already merged with main

File: terraform/src/dataFilterLambda/dataFilterLambda.js [hunk 9]
```diff
@@ -276,7 +277,7 @@ export const handler = async () => {
     );
     // Format into a csv string to allow reading into bucket
     const combinedImdGridallFileString = generateCsvString(
-      `POSTCODE,POSTCODE_2,LOCAL_AUT_ORG,NHS_ENG_REGION,SUB_ICB,CANCER_REGISTRY,EASTING_1M,NORTHING_1M,LSOA_2011,MSOA_2011,CANCER_ALLIANCE,ICB,OA_2021,LSOA_2021,MSOA_2021,IMD_RANK,IMD_DECILE`,
+      `POSTCODE,POSTCODE_2,LOCAL_AUT_ORG,NHS_ENG_REGION,SUB_ICB,CANCER_REGISTRY,EASTING_1M,NORTHING_1M,LSOA_2011,MSOA_2011,CANCER_ALLIANCE,ICB,OA_2021,LSOA_2021,MSOA_2021,IMD_RANK,IMD_DECILE,LSOA_NAME`,
```
Comment (Abdul-NHS) [inline terraform/src/dataFilterLambda/dataFilterLambda.js:280]: change already in main, environment rename is fine

File: terraform/src/getLsoaInRangeLambda/getLsoaInRangeLambda.js [hunk 10]
```diff
@@ -145,38 +158,39 @@ export async function scanLsoaTable(client, lastEvaluatedItem, tableItems) {
...
-      console.log("Table is larger than 1Mb hence recursively routing through to obtain all data"
+      console.log(
+        "Table is larger than 1Mb hence recursively routing through to obtain all data"
       );
       tableItems.push(response.Items);
       lastEvaluatedItem = response.LastEvaluatedKey;
       await scanLsoaTable(client, lastEvaluatedItem, tableItems);
     } else {
-      console.log("Unsuccess")
+      console.log("Unsuccess");
```
Comment (Abdul-NHS) [inline terraform/src/getLsoaInRangeLambda/getLsoaInRangeLambda.js:170]: linting changes that should be reverted until we decide on a linter as a team

**Review by zainm97 — DISMISSED** [review]

**Review by zainm97 — APPROVED** [review]

File: scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.js (hunk 3 above)
Comment (Abdul-NHS) [inline scripts/test_data/generateLsoaMidpoint/generateLsoaMidpoint.js:50]: still included the changes from main, however the commit is included so not sure whats going on. However will approve

**Review by zainm97 — DISMISSED** [review]

**Review by zainm97 — APPROVED** [review]

**Review by Abdul-NHS — APPROVED** [review]

**Review by zainm97 — DISMISSED** [review]
